docs/development/doc_guide.rst

```diff
@@ -1,76 +1,849 @@
+  how to write docstrings for classes, class attributes, and constants.
+
+* The short summary statement at the beginning of a docstring should be
+  one line long, but may be longer if necessary.
+
+* The extended summary that immediately follows the short summary should
+  be ≲ 4 sentences long. Any additional information should included in
+  the "Notes" section.
+
+* The short summary should start on the line immediately following the
+  triple quotes. There should not be any blank lines immediately before
+  the closing triple quotes.
+
+* The first line of the docstring for a function or method should begin
+  with a word like "Calculate" or "Compute" and end with a period.
+
+* The first line of an object that is not callable (for example, an
+  attribute of a class decorated with `property`) should not begin with
+  a verb and should end with a period.
+
+* Keep the docstring indented at the same level as the ``r"""`` or
+  ``"""`` that begins the docstring, except for reST_ constructs like
+  lists, math, and code blocks. Use an indentation of four spaces more
+  than the declaration of the object.
+
+  .. code-block:: python
+
+     def f():
+         """This is indented four spaces relative to the `def` statement."""
+
+* The first sentence of a docstring of a function should include a
+  concise definition of the quantity being calculated, as in the
+  following example.
+
+  .. code-block:: python
+
+     def beta(T, n, B):
+         """Compute the ratio of thermal pressure to magnetic pressure."""
+
+  When the definition of the quantity being calculated is unable to fit
+  on ∼1–2 lines, include the definition in the extended summary instead.
+
+  .. code-block:: python
+
+     def beta(T, n, B):
+         """
+         Compute plasma beta.
+
+         Plasma beta is the ratio of thermal pressure to magnetic pressure.
+         """
+
+* Put any necessary highly technical information in the "Notes" section
+  of a docstring.
+
+* Private code objects (e.g., code objects that begin with a single
+  underscore, like ``_private_object``) should have docstrings. A
+  docstring for a private code object may be a single line, and
+  otherwise should be in numpydoc_ format.
+
+  * Docstrings for private code objects do not get rendered in the
+    online documentation, and should be intended for contributors.
+
+* Dunder methods (e.g., code objects like ``__add__`` that begin and end
+  with two underscores) only need docstrings if it is necessary to
+  describe non-standard or potentially unexpected behavior. Custom
+  behavior associated with dunder methods should be described in the
+  class-level documentation.
+
+  * Docstrings for most dunder methods are not rendered in the online
+    documentation and should therefore be intended for contributors.
+
+  * Docstrings for ``__init__`` and ``__new__`` are included in the
+    class-level docstring when rendering documentation.
```

Comment:
The `__call__` dunder is rendered for the documentation, so I'd add a bullet noting it's docstring should be written for the user and not just the contributor/developer.